paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/CustomerAdapter.kt

```diff
@@ -0,0 +1,150 @@
+package com.stripe.android.paymentsheet.repositories
+
+import android.content.Context
+import androidx.annotation.RestrictTo
+import com.stripe.android.ExperimentalSavedPaymentMethodsApi
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * [CustomerAdapter] A "bridge" from wallet mode to your backend to fetch Customer-related
+ * information. Typically, you will not need to implement this interface yourself. You should
+ * instead use [CustomerAdaper.create], which manages retrieving and updating a Stripe customer for
+ * you.
+ *
+ * Implement this interface if you would prefer retrieving and updating your Stripe customer object
+ * via your own backend instead of using the default implementation.
+ *
+ * Use [CustomerAdapter.create] to create an instance of a [CustomerAdapter].
+ */
+@ExperimentalSavedPaymentMethodsApi
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface CustomerAdapter {
+    /**
+     * Retrieves a list of payment methods attached to the customer provided by the
+     * [CustomerEphemeralKeyProvider]
```

Comment:
Hmm, `CustomerEphemeralKeyProvider` is an implementation detail as well.

I'm wondering if we should add some more docs to our factory, and make these more generic. But I see the value in leaving some of these here as well.

Reply:
Yea, this can be more generic, since it is up to the implementer